codersdk/workspacesdk/workspacesdk.go

```diff
@@ -392,6 +389,12 @@ func (c *Client) AgentReconnectingPTY(ctx context.Context, opts WorkspaceAgentRe
 	return websocket.NetConn(context.Background(), conn, websocket.MessageBinary), nil
 }
 
+func WithTestOnlyCoderContextResolver(ctx context.Context, r Resolver) context.Context {
```

Comment:
This needs to be exported so that CLI tests can mock/fake the resolver. Now that there is a way to do it from outside the package, we no longer need `workspacesdk_internal_test.go` to be internal, so those tests are moved to the existing `workspacesdk_test.go`